odk1-src/form-functions.rst

```diff
@@ -0,0 +1,674 @@
+
+  The identifiers list in :func:`format-date` are available, 
+  plus the following:
+  
+  .. csv-table::
+  
+    %H, 0-padded hour (24-hr time)
+    %h, hour (24-hr time)
+    %M, 0-padded minute
+    %S, 0-padded second
+    %3, 0-padded millisecond ticks.
+
+.. _geography-functions:
+    
+Geography
+------------
+
+.. function:: area(nodeset | geoshape) 	
+
+  Returns the area, in square meters, 
+  of either a :arg:`nodeset` of geopoints or a :arg:`geoshape` value.
+  
+  It takes into account the circumference of the Earth around the Equator but does not take altitude into account.
+
+.. function:: distance(nodeset | geoshape | geotrace)
+
+  Returns the distance, in meters, of either:
+  
+  - a :arg:`nodeset` of geopoints
+  - the perimeter of a :arg:`geoshape`
+  - the length of a :arg:`geotrace` value
+  
+  It takes into account the circumference of the Earth around the Equator and does not take altitude into account.
+
+.. _utility-functions:
+
+Utility
+---------
+  
+.. function:: position(xpath)
+
+  Returns an integer equal to the 1-indexed position of the current node
+  within the node defined by :arg:`xpath`.
+  
+  Most often this is used in the form :tc:`position(..)`
+  to identify the current iteration index
+  within a repeat group.  
+  
+  .. include:: incl/form-examples/parallel-repeat-groups.rst
+
+.. function:: uuid([length]) 	
+
+  Without argument, returns a random `RFC 4122 version 4 compliant UUID`__. 
+  
+  __ https://en.wikipedia.org/wiki/Universally_unique_identifier#Version_4_(random)
+  
+  With an argument it returns a random GUID of specified :arg:`length`.  
+
+.. function:: once(expression)
+
+  Returns the value :arg:`expression` if the question's value is empty. Otherwise, returns the current value of the question.
+
+  This can be used to ensure that a random number is only generated once,
+  or to store the first value entered for a question
+  in a way that is retrievable even of the response is changed later.
+
+.. function:: random()
+
+  Returns a random number between 0.0 (inclusive) and 1.0 (exclusive).
+
+  
+.. function:: randomize(nodeset[, seed]) 	
+
+  Returns a shuffled :arg:`nodeset`.
+  
+  A shuffle with a numeric :arg:`seed` is deterministic and reproducible.
+  
+.. function:: checklist(min, max, response[, response[, response [, ... ]]])
+
+  Returns ``True`` if the number of :arg:`response`\ s that are exactly the string "yes" is between :arg:`min` and :arg:`max`, inclusive.  
+  
+  Set :arg:`min` or :arg:`max` to ``-1`` to make the argument not applicable.
+
+.. function:: weighted-checklist(min, max, reponse, weight[, response, weight[, response, weight[, response, weight[, ... ]]])
+
+  Returns ``True`` if 
+  the sum of the :arg:`weight`\ s 
+  of each :arg:`response` that is exactly the string "yes"
+  is between :arg:`min` and :arg:`max`, inclusive.
+  
+  Set :arg:`min` or :arg:`max` to ``-1`` to make the argument not 
+    
+
+.. _xform-operators:
```

Comment:
I really like this section. Nice and clear. But it's the most common thing people would look for, so I think it should go first.

And now that I've written that, maybe it should just be in its own section after Form Logic